tests/end2end/playwright/google-basemap.spec.js

```diff
@@ -0,0 +1,14 @@
+
+const { test, expect } = require('@playwright/test');
+const { gotoMap } = require('./globals');
+
+test.describe('Google Maps Baselayers', () => {
+    test('Load map', async ({ page }) => {
+        const url = '/index.php/view/map?repository=testsrepository&project=google_basemap';
+        await gotoMap(url, page);
+
+        // baselayers group should be hidden since it is empty due to the default STRICT_GOOGLE_TOS_CHECK env variable value = TRUE
```

Comment:
For testing purpose, maybe we can switch to `False` in the docker-compose file, or add a dummy Google API key in the CFG file, so the transformation to OL Google layer can be tested ?

Reply:
Switching the docker-compose variable to `false` could lead to non predictable response, I don't know if the test could be solid this way, but it is an option. Adding a dummy API Key will raise a console error for sure, so it's predictable on 100%. 

If the purpose is to test the OL Google initialization I would choose the last one.

Another way is to duplicate the project by keeping the default `true` value for the `tos check`:
1. project wit no API key -> testing baselayers group is empty (no error in console)
2. project with dummy API Key -> testing OL class init

What do you think about it?

Reply:
> Adding a dummy API Key will raise a console error for sure, so it's predictable on 100%.

I though this would lead to empty tiles or similar (tiles with the Google Watermark saying "For dev purpose only")  or a JS warning, not error.

I though it would be good to test "your" actual Javascript content from this PR, so indeed the OL Google init from LWC.

It seems good what you propose.

Reply:
> I though this would lead to empty tiles or similar

Sort of. I can confirm that OL class requires a valid API Key to create the session. The is no option to get a dummy api key only for testing purpose to get things works with no errors. 

![image](https://github.com/3liz/lizmap-web-client/assets/20710470/573cf606-adef-408c-9061-404f2dd4d3ea)

I can try to catch these errors to test that the system tries to start the Google session issuing a generic APY key

Reply:
Ok, thanks for checking.

> I can try to catch these errors to test that the system tries to start the Google session issuing a generic APY key

Ideally, yes, if you want to improve and warn

So let's merge like this ?

Reply:
For me yes